modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb

```diff
@@ -253,25 +247,21 @@ def get_completed_phase_number_from_phase_details(details)
         end
 
         def get_bgs_phase_completed_dates(data)
-          phase_dates = {}
-
-          if data&.dig(:benefit_claim_details_dto, :bnft_claim_lc_status).is_a?(Array)
-            max_completed_phase = 0
-            data[:benefit_claim_details_dto][:bnft_claim_lc_status].each_with_index do |lc, i|
-              completed_phase_number = get_completed_phase_number_from_phase_details(lc)
-              if i.zero?
-                max_completed_phase = completed_phase_number
-                phase_dates["phase#{completed_phase_number}CompleteDate"] = date_present(lc[:phase_chngd_dt])
-              elsif completed_phase_number.present? && completed_phase_number < max_completed_phase
-                phase_dates["phase#{completed_phase_number}CompleteDate"] = date_present(lc[:phase_chngd_dt])
+          lc_status_array =
+            [data&.dig(:benefit_claim_details_dto, :bnft_claim_lc_status)].flatten.compact
+          # compact eliminates nils
+          max_completed_phase = lc_status_array.first[:phase_type_change_ind].split('').first
+          return {} if max_completed_phase.downcase.eql?('n')
+
+          {}.tap do |phase_date|
+            lc_status_array.reverse.map do |phase| # reverse to ascend so we get the more recent completed
+              completed_phase_number = phase[:phase_type_change_ind].split('').first # grab phase ind
+              if completed_phase_number <= max_completed_phase &&
+                 completed_phase_number.to_i.positive? # a string will return 0 here
```

Comment:
the comments were intended to explain my logic in the example snippet from slack. I didn't intend these to be included. I don't think they serve much purpose here.